### DISCUSSION TITLE
PageUp and PageDown Keys (in 4 directional layouts)

### DISCUSSION THREAD

**Discussion by sifran-github** [freeplane/freeplane#1081] [General]
In continuation of this issue report https://github.com/freeplane/freeplane/issues/1078, I create this thread to ask the community:
**How do you think Freeplane should handle PageUp and PageDown keys, in horizontal layouts?**

Until the new version, when we had only 2 layout directions, and pressing these keys did "Go to first/Last siblings".
But now we add 2 more directions. This is why the question is being asked.

For example, if the selected node is this ("LLL"), and you press Page Up or Page down. Where should the selection go?
![image](https://user-images.githubusercontent.com/5730355/230460318-6c06fbb3-5b0e-4da9-91e7-8a90888f6f61.png)

Should these keys behave by FUNCTION (select the first/last siblings which are now on the left and right sides of the selected node) or by DIRECTION (going up and down somewhere)?

-------------

I create this separate from the big thread (https://github.com/freeplane/freeplane/discussions/989) about the arrow keys (up/down/left/right), because:

1. That discussion is already long.
2. After a lot of changes and improvements, it looks like the arrow keys are working well. I personally tested it for a few hours at least (just pressing up/down/left/right in many layout configurations) and i'm very impressed how good it is.

-------------

Although in case of the arrow keys, the decision was to have it based on DIRECTION, my initial thought is to keep PageUp and PageDown to behave by FUNCTION. Your thoughts?

-------------

Notes:
- I also notice that we don't utilize HOME and END keys for map movement, but rather to edit the selected node, which can be done anyway by any key or by using F2. Maybe it can help with some ideas. We can keep it like that, its just some observation that might help to bring some ideas.
- In older versions, PageUp and PageDown are working but are not in any menu or in the "help->key reference" list.

**Comment by EdoFro** [comment]
Go to first/Last siblings

**Comment by dpolivaev** [comment]
I fixed the navigation in the last preview.
PageUp and PageDown are not in any menu and therefore not in the "help->key reference" list. And I could add them to both places simultaneously (but not to only one of them without the other one).
They should behave "by function" because the "home" and "end" keys start editing the node at the first/last character.
